### PR TITLE
Allow Database Hosts to have multiple Nodes

### LINKED ISSUE #133: Make DatabaseHosts available to multiple (many) Nodes

### PR DESCRIPTION
**Pull request by alexevladgabriel**
Closes #133

### CODE REVIEW

File: database/migrations/2024_10_31_203540_change_database_hosts_to_belong_to_many_nodes.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('database_host_node', function (Blueprint $table) {
+            $table->id();
```
Comment (RMartinOscar) [inline database/migrations/2024_10_31_203540_change_database_hosts_to_belong_to_many_nodes.php:12]:
```suggestion
            $table->id();
            $table->unsignedInteger('node_id');
```

File: database/migrations/2024_10_31_203540_change_database_hosts_to_belong_to_many_nodes.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Schema::table('database_hosts', function (Blueprint $table) {
+            $table->dropForeign(['node_id']);
+            $table->dropColumn('node_id');
+        });
+    }
+
+    public function down(): void
+    {
+        Schema::table('database_hosts', function (Blueprint $table) {
+            $table->foreignId('node_id')->nullable()->references('id')->on('nodes');
```
Comment (RMartinOscar) [inline database/migrations/2024_10_31_203540_change_database_hosts_to_belong_to_many_nodes.php:37]:
```suggestion
            $table->unsignedInteger('node_id')->nullable();
            $table->foreignId('node_id')->references('id')->on('nodes');
```

**Review by RMartinOscar — CHANGES_REQUESTED** [review]
MySql/Mariadb needs explicit type

**Comment by notAreYouScared** [comment]
Closed in favor of #767

**Pull request closed without merging**